cadence/transactions/nftStorefront/purchase_listing.cdc

```diff
@@ -53,11 +53,11 @@ transaction(listingResourceID: UInt64, storefrontAddress: Address) {
 
     execute {
         let item <- self.listing.purchase(
-            payment: <-self.paymentVault
+            payment: <-self.paymentVault,
+            commissionRecipient: nil
         )
 
         self.kittyItemsCollection.deposit(token: <-item)
-
-        self.storefront.cleanup(listingResourceID: listingResourceID)
+        self.storefront.removeListing(listingResourceID: listingResourceID)
```

Comment:
That is... unfortunate.... If the only way we can remove a listing from the chain is through the `cleanupExpiredListings()` pub fun....

a suggestion off the top of my head could be to periodically call `cleanupExpiredListings` from 0 - total supply and allow listings to be cleaned up naturally but i can see how the worker can get costly very quickly. Maybe ask the team responsible for the v2 contract about how they want ppl to handle the scenario of a purchased listing? I feel like this is not an uncommon situation.

The public interface suggests to me they want ppl to have a cleanup worker to remove listings once they naturally expire. (assuming you cant update the expiry once a listing is created)

Reply:
@codingone21, you can use this one https://github.com/onflow/nft-storefront/blob/c0ef786ed94c5fc5f81e96d465aef7b9f4c6b35f/contracts/NFTStorefrontV2.cdc#L515 now. I have updated the smart contract on mainnet as well.

Reply:
Nice! Thank you @satyamakgec I will update the code accordingly :)